saml-core/src/test/java/org/keycloak/saml/processing/core/parsers/saml/SAMLParserTest.java

```diff
@@ -75,6 +84,29 @@ public void testSaml20EncryptedAssertionsSignedReceivedWithRedirectBinding() thr
     }
 
     @Test
+    public void testAsml20EncryptedAssertionWithNewlines() throws Exception {
+        try (InputStream st = SAMLParserTest.class.getResourceAsStream("KEYCLOAK-4489-encrypted-assertion-with-newlines.xml");
+             InputStream privateKeyStream = SAMLParserTest.class.getResourceAsStream("example.private.key")) {
+            Object parsedObject = parser.parse(st);
+            assertThat(parsedObject, instanceOf(ResponseType.class));
+
+            ResponseType resp = (ResponseType) parsedObject;
+            assertThat(resp.getAssertions().size(), is(1));
+
+            ResponseType.RTChoiceType rtChoiceType = resp.getAssertions().get(0);
+            assertNull(rtChoiceType.getAssertion());
+            assertNotNull(rtChoiceType.getEncryptedAssertion());
+
+            PrivateKey privateKey = PemUtils.decodePrivateKey(StreamUtil.readString(privateKeyStream, StandardCharsets.UTF_8));
```

Comment:
To minimize number of files around, can you please declare the keys as a constant instead of loading them from a separate files?